packages/ra-core/src/auth/useCanAccessRecordSources.ts

```diff
@@ -0,0 +1,152 @@
+import {
+    useQueries,
+    UseQueryOptions,
+    UseQueryResult,
+} from '@tanstack/react-query';
+import useAuthProvider from './useAuthProvider';
+import useLogoutIfAccessDenied from './useLogoutIfAccessDenied';
+import { useEvent } from '../util';
+import { useEffect } from 'react';
+import { useResourceContext } from '../core';
+
+export const combine = <ErrorType>(
```

Comment:
please move this function down to make the exported function the first in the file - it's more expected. 

Also, the name "combine" is too broad IMHO. What do we combine? Into what?